File: app/src/main/java/com/chaitupenjudcoder/firebasehelpers/BucksInputValidationHelper.java
package com.chaitupenjudcoder.firebasehelpers;

import android.support.design.widget.TextInputLayout;

public class BucksInputValidationHelper {

    public boolean inputValidator(TextInputLayout layout, int length) {
        String input = layout.getEditText().getText().toString();
        if (input.isEmpty()) {
            layout.setError("Field can\'t be empty");
            return false;
        } else if (input.length() > length){
            layout.setError("Field " + layout.getHint() + "\'s length can\'t be greater than " + length);
            return false;
        }else {
            layout.setError(null);
            return true;
        }
    }

    public boolean passwordValidator(TextInputLayout first, TextInputLayout second) {
        String firstInput = first.getEditText().getText().toString().trim();
        String secondInput = second.getEditText().getText().toString().trim();
        if (firstInput.equals(secondInput)) {
            second.setError(null);
            return true;
        } else {
            second.setError("Password and Confirm Password must match");
            return false;
        }
    }
}

File: app/src/main/java/com/chaitupenjudcoder/firebasehelpers/BucksInputValidationHelper.kt
package com.chaitupenjudcoder.firebasehelpers

import com.google.android.material.textfield.TextInputLayout

class BucksInputValidationHelper {
    fun inputValidator(layout: TextInputLayout, length: Int): Boolean {
        val input = layout.editText!!.text.toString()

        return when {
            input.isEmpty() -> {
                layout.error = "Field can\'t be empty"
                false
            }

            input.length > length -> {
                layout.error = "Field " + layout.hint + "\'s length can\'t be greater than " + length
                false
            }

            else -> {
                layout.error = null
                true
            }
        }
    }

    fun passwordValidator(first: TextInputLayout, second: TextInputLayout): Boolean {
        val firstInput = first.editText?.text.toString().trim { it <= ' ' }
        val secondInput = second.editText?.text.toString().trim { it <= ' ' }

        return if (firstInput == secondInput) {
            second.error = null
            true
        } else {
            second.error = "Password and Confirm Password must match"
            false
        }
    }
}